go/libraries/doltcore/sqle/dsess/autoincrement_tracker.go

```diff
@@ -398,7 +437,31 @@ func (a *AutoIncrementTracker) AcquireTableLock(ctx *sql.Context, tableName stri
 	return a.mm.Lock(tableName), nil
 }
 
-func (a *AutoIncrementTracker) InitWithRoots(ctx context.Context, roots ...doltdb.Rootish) error {
+func (a *AutoIncrementTracker) waitForInit() error {
+	done := make(chan struct{})
```

Comment:
Probably don't want to initialize a new channel here, this is called a lot.

I think instead I would get rid of the waitgroup entirely (it's just a single work item) and use a channel in the struct, then close the channel instead of calling `wg.Done()`. Your logic for waiting below still works fine.